### ISSUE TITLE
Fosterer: show upcoming fosters in the fostered pets page

### ISSUE THREAD

**Issue by kasugaijin** [rubyforgood/homeward-tails#1231]
Log in as a fosterer and go to `/adopter_fosterer/fostered_pets`. It currently only shows 'current' fosters on this page (create a foster as staff that is current and you will see it). We should also show upcoming fosters as well on this page. Let's stick with the small cards for now, but let's show the 'current' or 'upcoming' badge on the card next to the dog's name. 

See the staff/fosters page for the actual 'upcoming' and 'current' badges we use for fosters.

![image](https://github.com/user-attachments/assets/7d17f5ac-3829-4719-a8ef-6b1365ec6866)

**Comment by MichaScant** [comment]
Hey, can I work on this issue please? Thank you

**Comment by kasugaijin** [comment]
@MichaScant all yours

**Comment by MichaScant** [comment]
Hi, for some reason i'm unable to connect to the database after pulling from main, has there been a change recently that requires me to change something?

**Comment by MichaScant** [comment]
<img width="731" alt="Image" src="https://github.com/user-attachments/assets/9ad3128c-b7b7-455b-93de-b9623de5a02f" />

getting this error specifically ^

**Comment by kasugaijin** [comment]
Yeah there have been a few changes! Are you running locally?
I think this section in the README should help you fix this: 
```
If you have already set up this application locally, and are now getting errors related to Queue or Cable database hosts, please update your config/application.yml to specify the host as localhost for these databases (see config/application.example.yml).
```

**Comment by MichaScant** [comment]
I am running on docker, found a thread on slack that helped me solve one problem but now the password authentication is failing (both with or without application.yml)

**Comment by MichaScant** [comment]
<img width="1015" alt="Image" src="https://github.com/user-attachments/assets/d7d7c0a3-e51a-4470-a13d-8d53f1b2343a" />

it seems its trying to connect to 5432, however i noticed we are now using 54320, been playing around with the docker-compose trying to get it to work but no such luck so far

**Comment by MichaScant** [comment]
update: 

managed to get it to work by resetting the password for user postgres and removing application.yml, this has caused another issue where i can't access localhost3000/alta for some reason:

<img width="798" alt="Image" src="https://github.com/user-attachments/assets/f4d606e1-8b11-40c4-994b-7ef20c0dea42" />

**Comment by MichaScant** [comment]
update: solved the issue, had to go into the bash script and run rails db:setup manually, for some reason docker-compose run --rm app bin/setup has no affect now

**Comment by kasugaijin** [comment]
> the weird thing is I'm able to access it manually through this command without providing a password: docker exec -it pet-rescue-postgres-1 psql -U postgres
> 
> for context this is what my application.yml file has
> 
> <img alt="Image" width="462" src="https://private-user-images.githubusercontent.com/89426143/403990176-6bae1cf2-3b2d-4d03-ac87-4cfbe145cea3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwNTk5MjYsIm5iZiI6MTczNzA1OTYyNiwicGF0aCI6Ii84OTQyNjE0My80MDM5OTAxNzYtNmJhZTFjZjItM2IyZC00ZDAzLWFjODctNGNmYmUxNDVjZWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE2VDIwMzM0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViYzBkNWI5MTdlYzZhNWMxNDU4YTY4M2M4ZDgyNzY4NzczNzA1NTVmYmVmOWMzM2RkOTIzN2MyMTVhMmNkMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.de81OMO7Y_UMbvlzVzvK8ToSfH5Q-e2iCcuQhxBxo2M">

Ah. I run in Docker and my `application.yml` is empty because we handle that in the config files for Docker.

**Comment by kasugaijin** [comment]
> update: solved the issue, had to go into the bash script and run rails db:setup manually, for some reason docker-compose run --rm app bin/setup has no effect now

Yeah I don't use the bin folders in docker myself. I think they no longer work, as well, after recent changes. I just run things manually in the bash e.g. to drop, create, migrate, seed the db.

**Comment by MichaScant** [comment]
Ok, I see, I'll remember that for the future, thank you!

**Comment by MichaScant** [comment]
<img width="1423" alt="Image" src="https://github.com/user-attachments/assets/dd00bdce-5778-47ef-9295-27f08949098e" />

Hello, sorry for the delay, let me know if this look good and ill open a pr

**Comment by kasugaijin** [comment]
@MichaScant The placement looks good. We actually have badges for this that we can use! Check out the fosters page (log in as staff) and create some fosters and you will see the badges...or check out that view and you'll see the logic we use to create the badges.

**Comment by jmilljr24** [comment]
@cassieemb Assigned. Thanks for finishing these off.  Same as your previous issue, take a look at the previous pr/comments, but feel free to implement what you think is best!

**Comment by cassieemb** [comment]
@jmilljr24 Thank you! Do you have any preference on the placement of the badge now that the dates are added to the card as well? If not, I can play with the orientations and share a few options!

![Image](https://github.com/user-attachments/assets/4c66f46d-6d39-4b9d-9f67-3f40e69bc196)

**Comment by kasugaijin** [comment]
> [@jmilljr24](https://github.com/jmilljr24) Thank you! Do you have any preference on the placement of the badge now that the dates are added to the card as well? If not, I can play with the orientations and share a few options!
> 
> ![Image](https://github.com/user-attachments/assets/4c66f46d-6d39-4b9d-9f67-3f40e69bc196)

By badge do you mean the avatar? I think it looks good. The only thing that catches my eye is that maybe the pet name could be a little bit bigger?

**Comment by cassieemb** [comment]
> We actually have badges for this that we can use! Check out the fosters page (log in as staff) and create some fosters and you will see the badges for upcoming and current

I was referring to the badges for upcoming and current, but I can definitely make those changes as well!

**Comment by kasugaijin** [comment]
> > We actually have badges for this that we can use! Check out the fosters page (log in as staff) and create some fosters and you will see the badges for upcoming and current
> 
> I was referring to the badges for upcoming and current, but I can definitely make those changes as well!

Ah yes of course! Maybe above the dates if it’s not too squished? Have a play around and see what you think looks good.
